fix: minor edits to make dev docs a bit more clear

lwasser opened the pull request:
This also addresses #17374 . i ran into a bit of confusion setting up the dev environment. I thought I had to do it all using Docker but in the end I just discovered I can simply run make. I tried to make it clear to new contribs that they can skip the docker setup if they wish to contribute to the docs only!

di reviewed (approved):
Thanks!

woodruffw reviewed (commented):
Thanks a ton @lwasser!

lwasser commented on docs/dev/development/getting-started.rst:
```diff
@@ -30,8 +30,11 @@ improve the process:
 
-Getting the Warehouse source code
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Get the Warehouse source code
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
I *think* that the headers can be longer, but it will fail if they are shorter (I haven't used rust in a while so i could be totally wrong 🙃 ...). Let's merge, however. This did build locally for me!

di commented:
@lwasser I fixed the linting errors for you here since there isn't a way to do this automatically. (FWIW, my editor is configured to automatically strip trailing whitespace, you might want to do the same in your editor! I just added the backticks manually).

lwasser commented:
> @lwasser I fixed the linting errors for you here since there isn't a way to do this automatically. (FWIW, my editor is configured to automatically strip trailing whitespace, you might want to do the same in your editor! I just added the backticks manually).

Ahhhh ok. thanks!! i'll look into adding that step in my editor! i'm so used to pre-commit doing it for me it never occurred to me to set something like that up!
